### ISSUE TITLE
[CoE Starter Kit - QUESTION] Inactivity cleanup app doesn't show rejected and excused apps or flows

### ISSUE THREAD

**Issue by johnravas** [microsoft/coe-starter-kit#7753]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

The App and Flow Inactivity Notifications Clean Up app does not show any data in the Flows - rejected and excused view.  Same for the apps version.  I know I have flows and apps where the maker rejected the delete and I excused them from future evaluation.  I would like to be able to see this information.  I am also wondering if there is a good way or process to clear the "excused" status from apps and flows so they can be evaluated again.  In a perfect world, I'd love to be able to do this annually.

### What solution are you experiencing the issue with?

Governance

### What solution version are you using?

August 2023 3.11

### What app or flow are you having the issue with?

App and Flow Inactivity Notifications Clean Up

### What method are you using to get inventory and telemetry?

Cloud flows

AB#2412

**Comment by Jenefer-Monroe** [comment]
This filter does work for me. 
You can either excuse them from within the context of this app:
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/1ff76c14-c4f9-48b9-b17e-6ebf1d9af9af)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/470c227a-2a9b-438a-9f21-e1515957c3f3)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/ae94640f-95e1-42e3-a9f5-ed2d8d8451da)


Or you can excuse objects from inactivity in the Power Platform Admin View app
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/8971f0c7-ff35-47d5-ad9f-1e46295e172c)


Hope thats helpful!

**Comment by johnravas** [comment]
@Jenefer-Monroe the filter for apps and flows "rejected and not excused" works fine.  I have rejected and excused apps and flows but those views are not working.  See below.  I also just recently upgraded to the November edition.

![image](https://github.com/microsoft/coe-starter-kit/assets/59970962/4731cd88-5bec-45d6-8e2b-281dcc986747)

**Comment by Jenefer-Monroe** [comment]
Can you please show me how you rejected / excused the app?

**Comment by johnravas** [comment]
@Jenefer-Monroe Thanks for the response.  Here are the details:

Reject and excuse procedure:

1. Launch the App and Flow Inactivity Notifications Clean Up app
2. Go to the Flows - rejected buy not yet excused view
3. Click on a flow that appears in the view/gallery
4. Once in the screen that shows the details of the flow, click on the Mark flow as excused link/button
5. The flow disappears from that view
6. Go to the Flows - rejected and excused view
7. In this case, I am seeing the flow I just marked as excused!!

The problem is that I have lots of flows that I have excused.  When I look at the table directly I have lots of flows where the Flow Excused from Archive column is set to Yes.  Why aren't all of these previously marked flows showing in this view.

Let me set some context around why I am asking about this functionality.  As a part of our governance, when someone rejects the request to delete their flow, we have been marking them as excused - so they are not regularly pestered about deleting the flow.  The flaw in this logic, is that the flow is no longer evaluated to be deleted.  Eventually we want the flow to be re-evaluated and checked for deletion - just not every 30 days.  Getting our makers to edit and save the flow just to push off the check for 6 months is a stretch for our makers.  In a perfect world, after 1 year, would like to disable the excuse and make it subject to evaluation again.  I was hoping that the rejected and excused view might help us with that.  Otherwise, I need to go into the table and make changes.  Not a fan of doing that as it risks bypassing business logic.  Also there is no date assocated with the excuse, so I don't know how long it has been excused.

I hope this is helpful.  Looking forward to your response and guidance Jenefer.  Thank you!!

**Comment by Jenefer-Monroe** [comment]
Thats a totally reasonable workflow request. Once we sort out what is happening with you now, we can keep this issue for the upcoming feature work for a redo of this area.

In the meant time, I suspect these flows are excused, but do not have an active rejection in the archive table. 
Can you validate that?

**Comment by johnravas** [comment]
Absolutely, what would be the best way to validate that?

**Comment by johnravas** [comment]
In checking the Archive Approvals table, I see plenty of approvals and rejections with Approval Response Dates going back quite a ways - including the one I excused today.

**Comment by Jenefer-Monroe** [comment]
But the one you excused today is showing up in the view correct?

**Comment by johnravas** [comment]
Correct

**Comment by Jenefer-Monroe** [comment]
My theory is that you are expecting to see all the things you've excused in this view, but you will only see things in this view that that are both excused and which have a current rejection in that table. Those get deleted after a month so old ones wont have an entry. 
You can see all things you've excused in the PP Admin View by making a quick view, and then we can track this workflow for future work:
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/9de82ab7-fc4f-496b-bc2c-15d602e07fd9)

**Comment by johnravas** [comment]
OK that makes sense.  The one I excused recently has already disappeared.  Thoughts around my business case?  Is this something that could get in the queue for possible future consideration?

**Comment by Jenefer-Monroe** [comment]
yes absolutely this should be part of our thinking for this feature. Its not on the radar for a few months but then I plan to redo it personally. Will mark this one for that feature.

**Comment by johnravas** [comment]
Thank you Jenefer

**Comment by AmarSaiRam** [comment]
This issue is being closed as its scope is now included in a newly created consolidated enhancement request for centralized management of orphaned components (Apps, Flows, Connection References) in the CoE Starter Kit.

New Issue Reference: [[CoE Starter Kit - Feature]: https://github.com/https://github.com/microsoft/coe-starter-kit/issues/10329

Thank you for your contributions
